src/PatriciaTree.mli

```diff
@@ -0,0 +1,143 @@
+
+    {- The implementation uses Patricia Tree, as described in Okasaki
+      and Gill's 1998 paper
+      {{: https://www.semanticscholar.org/paper/Fast-Mergeable-Integer-Maps-Okasaki-Gill/23003be706e5f586f23dd7fa5b2a410cc91b659d}{i Fast mergeable integer maps}},
+      i.e. it is a space-efficient prefix trie over the big-endian representation of
+      the key's integer identifier.
+
+      Example of a 5-bit patricia tree containing five numbers: 0 [0b0000], 1 [0b0001],
+      5 [0b0101] and 7 [0b0111] and -8 [0b1111]:
+      {v
+                              Branch
+                          (prefix=0b?___)
+                          /             \
+                    Branch               Leaf(-8)
+                (prefix=0b0?__)          0b1111
+                /             \
+           Branch             Branch
+       (prefix=0b000?)     (prefix=0b01?_)
+         |        |          |       |
+      Leaf(0)  Leaf(1)    Leaf(5)  Leaf(7)
+      0b0000   0b0001     0b0101   0b0111
+      v}
+
+      The main benefit of Patricia Tree is that their representation
+      is stable (contrary to maps, inserting nodes in any order will
+      return the same shape), which allows different versions of a map
+      to share more subtrees in memory, and the operations over two
+      maps to benefit from this sharing. The functions in this library
+      attempt to maximally preserve sharing and benefit from sharing,
+      allowing very important improvements in complexity and running
+      time when combining maps or sets is a frequent operation.}
+
+    {- Finally, the implementation is more customizable, allowing
+      notably (key,value) pairs or different types to be in the same map,
+      or to choose the memory representation of the nodes of the tree.}
+
+    {- Some operations like {{!BASE_MAP.pop_unsigned_minimum}[pop_unsigned_minimum]} and
+     {{!BASE_MAP.pop_unsigned_maximum}[pop_unsigned_maximum]} make our Set
+     suitable as priority queue (but remember that each element in the
+     queue must map to a distinct integer, and that using the {{!unsigned_lt}unsigned order}
+     means elements with negative priority are seen as greater than elements with
+     positive ones).}
+    }
+
+    {b Note on complexity:} in the following, n represents the size of the
+    map when there is one (and [|map1|] is the number of elements in
+    [map1]).  The term log(n) correspond to the maximum height of the
+    tree, which is log(n) if we assume an even distribution of numbers
+    in the map (e.g. random distribution, or integers chosen
+    contiguously using a counter). The worst-case height is
+    O(min(n,64)) which is actually constant, but not really
+    informative; log(n) corresponds to the real complexity in usual
+    distributions. *)
+
+(** {1 Integer manipulations} *)
+
+include module type of Ints
+
+(** {1 Signatures} *)
+
+include module type of Signatures
+
+(** {1 Functors} *)
```

Comment:
Maybe say that functors are used to instantiate the library

Reply:
Well this is kind of the problem of this approach: where to document the files? Its either above the include, in the toplevel file, or at the top of the file itself... Since generating documentation just copies the `Functor.mli` file here, its a bit weird to have both.

I have admittedly, used different conventions here (`functors` is documented in the file, whereas `node` is documented at the include site). I'll try and unify those.